### PR TITLE
Calling functions with result of expression

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Realized that passing an expression as a param when calling a function wasn't working, i.e. calling foo(1 + 1) instead of foo(1)

This was broken in two spots. First was generating tokens, second was evaluating

Created tests that expose the behavior and came up with fixes. I'm somewhat worried that there are still a bunch of edge case type bugs lurking here.

### CODE REVIEW

File: apps/src/calc/expressionNode.js [hunk 1]
```diff
@@ -355,8 +359,9 @@ ExpressionNode.prototype.getTokenListDiff = function (other) {
  * @param {boolean} markDeepest Mark tokens in the deepest descendant
  */
 ExpressionNode.prototype.getTokenList = function (markDeepest) {
-  var depth = this.depth();
-  if (depth <= 1) {
+  if (!markDeepest || this.depth() <= 1) {
+    // if markDeepest is true, we diff against null so that everything is marked
```
Comment (bcjordan) [inline apps/src/calc/expressionNode.js]: Would this logic read a bit clearer if it were structured similar to how the comment explains it?

```
if (markDeepest) {
  // diff against null so everything is marked
} else {
  // diff against ourself, nothing will be marked
}

File: apps/src/calc/expressionNode.js [hunk 2]
```diff
@@ -166,7 +166,11 @@ ExpressionNode.prototype.evaluate = function (globalMapping, localMapping) {
       // We're calling a new function, so it gets a new local scope.
       var newLocalMapping = {};
       functionDef.variables.forEach(function (variable, index) {
-        var childVal = this.getChildValue(index);
+        var evaluation = this.children_[index].evaluate(globalMapping, localMapping);
```
Comment (bcjordan) [inline apps/src/calc/expressionNode.js]: Just for my own understanding, what's the difference between using `getChildValue` vs. evaluating the child at that index? Is it that the evaluated version will get processed much further?

**Comment by bcjordan** [comment]
LGTM, just a couple of small Qs but should be fine to merge

File: apps/src/calc/expressionNode.js (hunk 2 above)
Comment (Bjvanminnen) [inline apps/src/calc/expressionNode.js]: Yes. In the case of f(1 + 1), get childValue() returns '+', whereas evaluating gets us 2.

**Pull request merged**